src/parts/Portal/index.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { createPortal } from 'react-dom';
+
+interface PortalProps {
+  children: React.ReactNode;
+}
+
+const PORTAL_ROOT_ID = 'portal-root';
+
+const Portal = ({ children }: PortalProps): JSX.Element | null => {
+  const [portalElementNode, setPortalElementNode] = React.useState<HTMLElement>();
+
+  React.useEffect(() => {
+    const createPortalRootElement = () => {
```

Comment:
I would put this outside of the component body, as it doesn't have any dependencies on state.